cloudify_types/cloudify_types/component/component.py

```diff
@@ -0,0 +1,511 @@
+                                 timeout=self.timeout,
+                                 pollster_args=pollster_args,
+                                 expected_result=True):
+            return ctx.operation.retry(
+                'The deployment is not ready for execution.')
+
+        # we must to run some execution
+        if ((self.deployment.get(EXTERNAL_RESOURCE) and self.reexecute)
+                or not self.deployment.get(EXTERNAL_RESOURCE)):
+            execution_args = self.config.get('executions_start_args', {})
+            client_args = \
+                dict(deployment_id=self.deployment_id,
+                     workflow_id=self.workflow_id,
+                     **execution_args)
+            response = self.dp_get_client_response('executions',
+                                                   EXEC_START, client_args)
+
+            # Set the execution_id for the last execution process created
+            self.execution_id = response['id']
+            ctx.logger.debug('Executions start response: {0}'.format(response))
+
+            # Poll for execution success.
+            if not self.verify_execution_successful():
+                ctx.logger.error('Deployment error.')
+
+            ctx.logger.debug('Polling execution succeeded')
+
+        type_hierarchy = ctx.node.type_hierarchy
+
+        if DEP_TYPE in type_hierarchy:
+            ctx.logger.info('Start post execute component')
+            return self.post_execute_component()
+
+        raise NonRecoverableError(
+            'Unsupported node type provided {0}'.format(ctx.node.type))
+
+    def post_execute_component(self):
+        runtime_prop = ctx.instance.runtime_properties['deployment']
+        ctx.logger.debug(
+            'Runtime deployment properties {0}'.format(runtime_prop))
+
+        if 'outputs' \
+                not in ctx.instance.runtime_properties['deployment'].keys():
+            update_attributes('deployment', 'outputs', dict())
+            ctx.logger.debug('No component outputs exist.')
+
+        try:
+            ctx.logger.debug('Deployment Id is {0}'.format(self.deployment_id))
+            response = self.client.deployments.outputs.get(self.deployment_id)
+            ctx.logger.debug(
+                'Deployment outputs response {0}'.format(response))
+
+        except CloudifyClientError as ex:
+            _, _, tb = sys.exc_info()
+            raise NonRecoverableError(
+                'Failed to query deployment outputs: {0}'
+                ''.format(self.deployment_id),
+                causes=[exception_to_error_cause(ex, tb)])
+        else:
+            dep_outputs = response.get('outputs')
+            ctx.logger.debug('Deployment outputs: {0}'.format(dep_outputs))
+            for key, val in self.deployment_outputs.items():
+                ctx.instance.runtime_properties[
+                    'deployment']['outputs'][val] = \
+                    dep_outputs.get(key, '')
+        return True
```

Comment:
is `True` necessary? It doesn't seem to be used in a boolean manner.

Reply:
No, removed it